integration-tests/rpc_integration_test.go

```diff
@@ -0,0 +1,421 @@
+	require.NoError(t, err, "RPC server didn't start")
+
+	client, err := rpc.NewClient(standaloneRPCEndpointPrefix + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	// Create 10 new valid orders.
+	ethClient := ethclient.NewClient(ethRPCClient)
+	// NOTE(jalextowle): The default balances are not sufficient to create 10 valid
+	//                   orders, so we modify the zrx and weth amounts for this test
+	newWethAmount := new(big.Int).Div(wethAmount, big.NewInt(10))
+	newZrxAmount := new(big.Int).Div(zrxAmount, big.NewInt(10))
+	signedTestOrders := make([]*zeroex.SignedOrder, 10)
+	for i := 0; i < 10; i++ {
+		signedTestOrders[i] = scenario.CreateZRXForWETHSignedTestOrder(t, ethClient, makerAddress, takerAddress, newWethAmount, newZrxAmount)
+	}
+
+	// Send the newly created order to "AddOrders." The order is valid, and this should
+	// be reflected in the validation results.
+	validationResponse, err := client.AddOrders(signedTestOrders)
+	require.NoError(t, err)
+	assert.Len(t, validationResponse.Accepted, 10)
+	assert.Len(t, validationResponse.Rejected, 0)
+
+	// Send an initial "GetOrders" request through the rpc client. This request will
+	// get all of the orders in the database after the "AddOrders" request. We can
+	// test pagination by comparing to this list.
+	expectedPage := 0
+	expectedPerPage := 10
+	expectedSnapshotID := ""
+	initialGetOrdersResponse, err := client.GetOrders(expectedPage, expectedPerPage, expectedSnapshotID)
+	require.NoError(t, err)
+	assert.Len(t, initialGetOrdersResponse.OrdersInfos, expectedPerPage)
+
+	// Ensure that all of the orders that we added to the mesh node are represented in the
+	// get orders request.
+	for _, signedTestOrder := range signedTestOrders {
+		foundMatchingOrder := false
+		expectedOrderHash, err := signedTestOrder.ComputeOrderHash()
+		require.NoError(t, err)
+		signedTestOrder.ResetHash()
+
+		for _, orderInfo := range initialGetOrdersResponse.OrdersInfos {
+			if orderInfo.OrderHash.Hex() == expectedOrderHash.Hex() {
+				foundMatchingOrder = true
+				assert.Equal(t, signedTestOrder, orderInfo.SignedOrder, "signedOrder did not match")
+				assert.Equal(t, signedTestOrder.TakerAssetAmount, orderInfo.FillableTakerAssetAmount, "fillableTakerAssetAmount did not match")
+				break
+			}
+		}
+
+		assert.True(t, foundMatchingOrder, "found no matching entry in the getOrdersResponse")
+	}
+
+	// Make a new "GetOrders" request with different pagination parameters.
+	expectedPage = 1
+	expectedPerPage = 5
+	getOrdersResponse, err := client.GetOrders(expectedPage, expectedPerPage, expectedSnapshotID)
+	require.NoError(t, err)
+	assert.Len(t, getOrdersResponse.OrdersInfos, expectedPerPage)
+
+	// Ensure that the getOrdersResponse has the correct pagination
+	for i := range getOrdersResponse.OrdersInfos {
+		assert.Equal(t, initialGetOrdersResponse.OrdersInfos[i+5*expectedPage], getOrdersResponse.OrdersInfos[i], "Incorrect order of second getOrdersResponse")
```

Comment:
We don't really guarantee that the order of orders is the same between requests, it's just a side-effect of how orders are stored in the current database implementation. We shouldn't rely on this in our tests.

Instead of this, what do you think of sending two or three requests with pagination and then testing that _all_ the orders where returned (in any order)?

Reply:
That sounds like a good way to test for pagination. Is there a place that we document this guarantee?